lib/readMaidFile.js

```diff
@@ -1,15 +1,44 @@
+const fs = require('fs')
 const parseMarkdown = require('./parseMarkdown')
-const loadFile = require('./loadFile')
-
-module.exports = section => {
-  const { path: filepath, data } = loadFile.loadSync([
-    'maidfile.md',
-    'contributing.md',
-    'CONTRIBUTING.md',
-    'README.md',
-    'readme.md'
-  ])
-  if (!filepath) return null
-
-  return parseMarkdown(data, { section, filepath })
+const find = require('find-file-up')
+
+// a bit duplication with what we have in parseMarkdown.js, but we can't DRY it
+const MAIDFILE = 'maidfile.md'
+
+/**
+ * Resolving mechanism. Always starts to search for `maidfile.md` from
+ * the current working directory to 5 levels upwards (probably enough).
+ * Then repeats this for `.maidfile.md` (with dot) but 10 levels upwards,
+ * because who knows, someone may have big folder trees.
+ *
+ * Second step. If it can't find the Maid files (above ones) and there is
+ * no given `--config-path / -c` flag on the CLI it throws with error message
+ * thrown from the main Maid class constructor.
+ *
+ * Third step. If config path is given, it treats that config file
+ * a bit more specifically, as described in the (current) readme - task names should be
+ * h3 headers and should have some h2 header, which in turn can also be defined on
+ * the CLI through the `--section` flag.
+ *
+ * @param {Object} opts command line global flags
+ * @returns {Object} like `{ filepath, tasks }`, coming from parseMarkdown
+ */
+module.exports = (opts = {}) => {
+  let filepath =
+    find.sync(MAIDFILE, null, 5) || find.sync(`.${MAIDFILE}`, null, 10)
```

Comment:
What about simply using `loadFile`? we can just stop searching files when reaching `/`.

```js
loadFile.loadSync(['maidfile.md', '.maidfile.md'])
```

And remove the `stopDir` here: https://github.com/egoist/maid/blob/d2108b5bbe5a032885d2d1b7e35780e51a3e9603/lib/loadFile.js#L7